### DISCUSSION TITLE
Reading String Data (Group Codes 1000 and 1001)

### DISCUSSION THREAD

**Discussion by dkadish** [haplokuon/netDxf#406] [Q&A] [answered]
Hi,

I'm using the netDxf library and it's been excellent for importing DXF drawings to my application, but I'm having trouble finding the entry for what I am pretty sure is a custom (non-standard) data field. The software I'm writing deals with timber and the width of the timber used is stored using a pair of group codes 1001 and 1000 (see attached dxf for the full file. I have added the .txt extension so it would upload properly.).

Sometime after a `LWPOLYLINE` entry, the following appears:
```
1001
TimberData
1000
45x170 C24
```
The width data I'm looking for is the `45` part of the line `45x170 C24`, but I can't figure out how to access these lines using the API.

I am reading the document in using `DxfDocument dxfDoc = DxfDocument.Load(dxfFile);` and I am able to load and process the `LWPOLYLINE`s, but I could use some help finding this `TimberData`.

Thanks!

[T1.dxf.txt](https://github.com/haplokuon/netDxf/files/9357506/T1.dxf.txt)

**Comment by haplokuon** [comment, marked as answer]
What you are looking at is what is called Extended Data Information or XData for short. Make a search about "[extended data](https://github.com/haplokuon/netDxf/search?q=extended+data&type=issues)" you will find other users asking similar questions, or just look at issue #331. You can also read what the Autodesk documentation says about "[extended data](https://help.autodesk.com/view/OARX/2021/ENU/?guid=GUID-A2A628B0-3699-4740-A215-C560E7242F63)".

Or something like since you already know that there is only one entry in the XDataRecords and it is an string:
```c#
entity.XData.TryGetValue("TimberData", out XData xData);
string sectionData = xData?.XDataRecord[0].Value as string;
```
